src/tests/ftest/aggregation/dfuse_space_check.py

```diff
@@ -0,0 +1,164 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+import time
+from ior_test_base import IorTestBase
+
+
+class DfuseSpaceCheck(IorTestBase):
+    """Base Parallel IO test class.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DfuseSpaceCheck object."""
+        super(DfuseSpaceCheck, self).__init__(*args, **kwargs)
+        self.space_before = None
+        self.block_size = None
+
+    def tearDown(self):
+        """Tear down each test case."""
+        try:
+            if self.dfuse:
+                self.dfuse.stop()
+        finally:
+            # Stop the servers and agents
+            super(DfuseSpaceCheck, self).tearDown()
+
+    def get_nvme_free_space(self, display=True):
+        """Display pool free space.
+
+          Args:
+            display (bool): boolean to display output of free space.
+
+          Returns:
+            free_space_nvme (int): Free space available in nvme.
+        """
+        free_space_nvme = self.pool.get_pool_free_space("nvme")
+        if display:
+            self.log.info("Free nvme space: %s", free_space_nvme)
+
+        return free_space_nvme
+
+    def check_aggregation(self):
+        """Check Aggregation for 240 secs max, else fail the test."""
+        counter = 1
+        while self.get_nvme_free_space() != self.space_before:
+            # try to wait for 4 x 60 secs for aggregation to be completed or
+            # else exit the test with a failure.
+            if counter > 4:
+                self.log.info("Free space when test terminated: %s",
+                              self.get_nvme_free_space())
+                self.fail("Aggregation did not complete as expected")
+            time.sleep(60)
+            counter += 1
+
+    def write_multiple_files(self):
+        """Write multiple files
+
+          Returns:
+            file_count (int): Total number of files created before
+                              going out of space.
+        """
+        file_count = 0
+        while self.get_nvme_free_space(False) >= self.block_size:
+            file_loc = str(self.dfuse.mount_dir.value +
+                           "/largefile_{}.txt".format(file_count))
+            write_dd_cmd = u"dd if=/dev/zero of={} bs={} count=1".format(
+                file_loc, self.block_size)
+            if 0 in self.execute_cmd(write_dd_cmd, False, False):
+                file_count += 1
+
+        return file_count
+
+    def test_dfusespacecheck(self):
```

Comment:
The one large file test and the many small file test can be different test case if you think it's possible that some bug can cause only one of the cases to fail.